benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java

```diff
@@ -83,240 +78,127 @@
+   */
+  static Builder newBuilder(Option... supportedOptions) {
+    return new Builder(supportedOptions);
   }
 
-  static class Builder {
-
-    private final Set<Option> options;
-
-    private Builder() {
-      options = new LinkedHashSet<Option>();
-      options.add(HELP);
-    }
-
-    Builder addOptions(Option... opts) {
-      options.addAll(asList(opts));
-      return this;
-    }
-
-    void printUsage() {
-      System.out.println("Usage: [ARGS...]");
-      int maxWidth = 0;
-      for (Option option : options) {
-        maxWidth = max(commandLineFlag(option).length(), maxWidth);
-      }
-      // padding
-      maxWidth += 2;
-      for (Option option : options) {
-        StringBuilder sb = new StringBuilder();
-        sb.append(commandLineFlag(option))
-          .append(Strings.repeat(" ", maxWidth - sb.length()))
-          .append(option.description)
-          .append(option.required ? " Required." : "");
-        System.out.println("  " + sb);
-      }
-      System.out.println();
-    }
-
-    ClientConfiguration build(String[] args) {
-      ClientConfiguration config = new ClientConfiguration();
-      Set<Option> appliedOptions = new HashSet<Option>();
-
-      for (String arg : args) {
-        if (!arg.startsWith("--")) {
-          throw new IllegalArgumentException("All arguments must start with '--': " + arg);
-        }
-        String[] pair = arg.substring(2).split("=", 2);
-        String key = pair[0];
-        String value = "";
-        if (pair.length == 2) {
-          value = pair[1];
-        }
-        for (Option option : options) {
-          if (key.equals(option.toString())) {
-            if (option != HELP) {
-              option.action.applyNew(config, value);
-              appliedOptions.add(option);
-            } else {
-              throw new RuntimeException("");
-            }
-          }
-        }
-      }
-      for (Option option : options) {
-        if (option.required && !appliedOptions.contains(option)) {
-          throw new IllegalArgumentException("Missing required option '--" + option + "'.");
-        }
-      }
-      return config;
-    }
-
-    private static String commandLineFlag(Option option) {
-      return "--" + option + (option.type != "" ? '=' + option.type : "");
-    }
+  static class Builder extends AbstractConfigurationBuilder<ClientConfiguration> {
+    private ClientConfiguration config = new ClientConfiguration();
 
-    private interface Action {
-      void applyNew(ClientConfiguration config, String value);
+    private Builder(Option... supportedOptions) {
+      super(false, supportedOptionsSet(supportedOptions));
     }
 
-    enum Option {
-      HELP("", "Print this text.", new Action() {
-        @Override
-        public void applyNew(ClientConfiguration config, String value) {
+    @Override
+    protected void applyOption(Option option, String value) {
+      switch (option) {
+        case ADDRESS: {
```

Comment:
Should we get rid of all these blocks {} ? None of them are doing anything since none have local variables.